tests/rptest/tests/topic_delete_test.py

```diff
@@ -521,12 +565,18 @@ def topic_delete_unavailable_test(self, cloud_storage_type):
                    timeout_sec=30,
                    backoff_sec=1)
 
+        self._assert_topic_lifecycle_marker_status(
+            next_topic, LifecycleMarkerStatus.PURGED)
+
         # Eventually, the original topic should be deleted: this is the tiered
         # storage scrubber doing its thing.
         wait_until(lambda: self._topic_remote_deleted(self.topic),
                    timeout_sec=30,
                    backoff_sec=1)
```

Comment:
I'm surprised that `topic_remote_deleted` does not get tripped up by the uploaded life-cycle marker. I guess the `key_to_topic` method in `shared_client_utils.py` does not detect it. Would be nice to update it such that it picks up the lifecycle marker and explicitly ignore it somewhere else.